python/ray/tests/test_bounded_unix_sockets.py

```diff
@@ -0,0 +1,79 @@
+import sys
+import os
+
+import ray
+import logging
+import psutil
+
+import pytest
+
+logger = logging.getLogger(__name__)
+
+"""
+Detects number of sockets created by Core Workers.
+
+We need exact 2 unix sockets to Raylet: one for `raylet` and one for `plasma_store`.
+Example:
+
+    pconn(fd=31, family=<AddressFamily.AF_UNIX: 1>, type=<SocketKind.SOCK_STREAM: 1>, laddr='', raddr='/tmp/ray/session_2023-11-14_13-21-21_747315_27107/sockets/raylet', status='NONE'),  # noqa: E501
+    pconn(fd=78, family=<AddressFamily.AF_UNIX: 1>, type=<SocketKind.SOCK_STREAM: 1>, laddr='', raddr='/tmp/ray/session_2023-11-14_13-21-21_747315_27107/sockets/plasma_store', status='NONE')  # noqa: E501
+
+However it's a long overdue bug that we never closed FDs on raylet's fork to create core
+workers, making each core worker end up with O(N^2) open sockets. This test ensures this is fixed.
+
+However it's not that easy:
+
+- We can't determine Ray sockets by name: in Linux, it does not show the address as in
+    pconn.raddr, so we don't have a way to make sure it's the socket we are asserting.
+- We can't assert that len(all_sockets) == 2, because in MacOS, there are some
+    system-created sockets like '/var/run/mDNSResponder'.
+
+So we do it good old way (see python/ray/tests/test_actor_bounded_threads.py), test that
+number of sockets do not grow as time goes, across workers.
+"""
+
+
+def my_sockets():
+    return psutil.Process().connections("unix")
+
+
+def assert_bounded_sockets(prev_sockets, new_sockets):
+    print(new_sockets)
+    assert len(new_sockets) <= len(
+        prev_sockets
+    ), f"prev {prev_sockets}, new {new_sockets}"
+
+
+@ray.remote
+def fibonacci_and_assert_sockets(i):
```

Comment:
hmm test seems a little hard to understand. Is it impossible to just keep creating/killing actors instead? (and check # of sockets in actors doesn't grow)?

Reply:
added